homeassistant/helpers/entity_registry.py

```diff
@@ -591,8 +626,17 @@ def async_get_or_create(
                 unit_of_measurement=unit_of_measurement,
             )
 
+        entity_registry_id: str | None = None
+        deleted_entity = self.deleted_entities.get((domain, platform, unique_id))
+        if deleted_entity is not None:
+            self.deleted_entities.pop((domain, platform, unique_id))
+            # Restore id
+            entity_registry_id = deleted_entity.id
+
         entity_id = self.async_generate_entity_id(
```

Comment:
How is the entity_id restored when using a entity entry id of a deleted entity?

Reply:
The `entity_id` is not restored. We could add that in a follow-up if it's wanted. We can't guarantee restoring an `entity_id` though, or it would mean an `entity_id` can only be reused 30 days after it is no longer used.

Reply:
Ok.